File: java/src/main/java/problems101_200/TwoSum2InputArrayIsSorted.java
package problems101_200;

enum TwoSum2InputArrayIsSorted {
    DEFAULT {
        @Override
        int[] twoSum(int[] numbers, int target) {
            int start = 0;
            int end = numbers.length - 1;
            while (start < end) {
                int sum = numbers[start] + numbers[end];
                if (sum > target) {
                    end--;
                } else if (sum < target) {
                    start++;
                } else {
                    break; // found
                }
            }
            return new int[]{++start, ++end};
        }
    };

    abstract int[] twoSum(int[] numbers, int target);
}

File: kotlin/src/main/kotlin/problems101_200/TwoSum2InputArrayIsSorted.kt
package problems101_200

enum class TwoSum2InputArrayIsSorted {
    DEFAULT {
        override fun twoSum(numbers: IntArray, target: Int): IntArray {
            var start = 0
            var end = numbers.size - 1
            while (start < end) {
                val sum = numbers[start] + numbers[end]
                if (sum > target) {
                    end--
                } else if (sum < target) {
                    start++
                } else {
                    break // found
                }
            }
            return intArrayOf(++start, ++end)
        }
    },
    ;

    abstract fun twoSum(numbers: IntArray, target: Int): IntArray
}
